### PR TITLE
Add Kotlin samples

### PR DESCRIPTION
**Pull request by bellarm**
Code samples of the Google Maps Android API in Kotlin.

### CODE REVIEW

File: ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/BasicMapDemoActivity.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        val mapFragment : SupportMapFragment? =
+                supportFragmentManager.findFragmentById(R.id.map) as? SupportMapFragment
+        mapFragment?.getMapAsync(this)
+    }
+
+    /**
+     * This is where we can add markers or lines, add listeners or move the camera. In this case,
+     * we just move the camera to Sydney and add a marker in Sydney.
+     */
+    override fun onMapReady(googleMap: GoogleMap) {
```
Comment (jfschmakeit) [inline ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/BasicMapDemoActivity.kt]: Should this be `GoogleMap?` with the usual null check?

File: ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/CircleDemoActivity.kt [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+                    radiusMeters = centerMarker.position.distanceFrom(radiusMarker.position)
+                    circle.radius = radiusMeters
+                }
+                else -> return false
+            }
+            return true
+        }
+
+        fun onStyleChange() {
+            with(circle) {
```
Comment (jfschmakeit) [inline ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/CircleDemoActivity.kt]: Please add a comment here.

File: ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/CircleDemoActivity.kt [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+     */
+    override fun onMapReady(googleMap: GoogleMap?) {
+        map = googleMap ?: return
+        // we need to initialise map before creating a circle
+        with(map) {
+            moveCamera(CameraUpdateFactory.newLatLngZoom(sydney, 4.0f))
+            setContentDescription(getString(R.string.circle_demo_details))
+            setOnMapLongClickListener { point ->
+                // We know the center, let's place the outline at a point 3/4 along the view.
+                val view: View = supportFragmentManager.findFragmentById(R.id.map).view as View
```
Comment (jfschmakeit) [inline ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/CircleDemoActivity.kt]: Potential NPE if findFragmentById(..) doesn't match a fragment.

File: ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/CircleDemoActivity.kt [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+        // Set the strokeColorArgb if the SeekBars for it is changed, otherwise keep the old value
+        strokeColorArgb = when (seekBar) {
+            strokeHueBar -> Color.HSVToColor(Color.alpha(strokeColorArgb),
+                    floatArrayOf(progress.toFloat(), 1f, 1f))
+            strokeAlphaBar -> Color.argb(progress, Color.red(strokeColorArgb),
+                    Color.green(strokeColorArgb), Color.blue(strokeColorArgb))
+            else -> strokeColorArgb
+        }
+
+        circles.map { it.onStyleChange() }
```
Comment (jfschmakeit) [inline ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/CircleDemoActivity.kt]: Add a comment here.

File: ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/BasicMapDemoActivity.kt (hunk 1 above)
Comment (bellarm) [reply ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/BasicMapDemoActivity.kt]: Fixed in the latest commit

File: ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/UiSettingsDemoActivity.kt [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+     * but we are using EasyPermissions to handle it so we can suppress the "MissingPermission"
+     * check.
+     */
+    @SuppressLint("MissingPermission")
+    @AfterPermissionGranted(REQUEST_CODE_LOCATION)
+    private fun enableMyLocation() {
+        if (hasLocationPermission()) {
+            map.isMyLocationEnabled = true
+        } else {
+            EasyPermissions.requestPermissions(
```
Comment (katherineallen) [inline ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/UiSettingsDemoActivity.kt]: This function doesn't need each parameter on a new line - that is only for function declarations.

File: ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/CircleDemoActivity.kt (hunk 2 above)
Comment (bellarm) [reply ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/CircleDemoActivity.kt]: Fixed in the latest commit

File: ApiDemos/kotlin/app/build.gradle [hunk 6]
```diff
@@ -27,8 +27,16 @@ dependencies {
...
     implementation 'com.android.support.constraint:constraint-layout:1.0.2'
+    // This dependency is needed to use RecyclerView, for the LiteListDemoActivity
+    implementation "com.android.support:recyclerview-v7:27.0.2"
+    // CardView is used in the LiteListDemoActivity
+    implementation 'com.android.support:cardview-v7:27.0.2'
     testImplementation 'junit:junit:4.12'
     androidTestImplementation 'com.android.support.test:runner:1.0.1'
     androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
+    implementation 'com.google.android.gms:play-services-maps:11.8.0'
+    compile 'com.google.android.gms:play-services-location:11.8.0'
```
[the remark this answers is not in the document]
Comment (bellarm) [reply ApiDemos/kotlin/app/build.gradle]: Fixed in the latest commit

File: ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/UiSettingsDemoActivity.kt (hunk 5 above)
Comment (bellarm) [reply ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/UiSettingsDemoActivity.kt]: Fixed in the latest commit

**Pull request merged**